planet/clients/data.py

```diff
@@ -216,3 +307,207 @@ async def get_stats(self,
+        """Get an item type by id.
+
+        An item_type represents the class of spacecraft and/or processing level
+        of an item. All items have an associated item_type. Each item_type has
+        a set of supported asset_types which may be produced for a given item.
+
+        Parameters:
+            item_type_id: Item type identifier.
+
+        Returns:
+            Item type details.
+
+        Raises:
+            planet.exceptions.APIError: On API error.
+        """
+        raise NotImplementedError
+
+    async def get_item(
+        self,
+        item_type_id: str,
+        item_id: str,
+    ) -> dict:
+        """Get an item by id and item type id.
+
+        In the Planet API, an item is an entry in our catalog, and generally
+        represents a single logical observation (or scene) captured by a
+        satellite. Each item is defined by an item_type_id, which specifies the
+        class of spacecraft and/or processing level of the item. Assets (or
+        products, such as visual or analytic) can be derived from the item's
+        source data.
+
+        Parameters:
+            item_type_id: Item type identifier.
+
+        Returns:
+            Item details.
+
+        Raises:
+            planet.exceptions.APIError: On API error.
+        """
+        raise NotImplementedError
+
+    async def list_item_assets(self, item_type_id: str,
+                               item_id: str) -> typing.List[dict]:
+        """List all assets available for an item.
+
+        An asset describes a product that can be derived from an item's source
+        data, and can be used for various analytic, visual or other purposes.
+        These are referred to as asset_types.
+
+        Parameters:
+            item_type_id: Item type identifier.
+            item_id: Item identifier.
+
+        Returns:
+            Descriptions of available assets as a dictionary with asset_type_id
+            as keys and asset description as value.
+
+        Raises:
+            planet.exceptions.APIError: On API error.
+        """
+        raise NotImplementedError
+
+    async def get_asset(self,
+                        item_type_id: str,
+                        item_id: str,
+                        asset_type_id: str) -> dict:
+        """Get an item asset.
+
+        Parameters:
+            item_type_id: Item type identifier.
+            item_id: Item identifier.
+            asset_type_id: Asset type identifier.
+
+        Returns:
+            Description of the asset.
+
+        Raises:
+            planet.exceptions.APIError: On API error.
+            planet.exceptions.ClientError: If asset type identifier is not
+            valid.
+        """
+        # NOTE: this is not an API endpoint
+        # this is getting an asset by name from the dict returned by
+        # list_item_assets()
+        raise NotImplementedError
+
+    async def activate_asset(self, asset: dict):
```

Comment:
I went back and forth on having the input for asset functions be `asset` (description of the asset) or `[item_id, item_type_id, asset_type_id]`. The latter has the benefit of simpler parameters and not requiring the user to run `get_asset()`.

I ultimately went with `asset` because it will cut down on each function having to call `get_asset()`. One concern with this is that the user may be providing a description of an asset that is no longer in sync with what is on the server (an example being that the user provides a description of an asset that has a download location url, but the asset has since been inactivated so the download url is no longer valid).